### PR TITLE
feat: bring dependency injection container and config validation via zod

### PR DESCRIPTION
**Pull request by Plopix**
This PR brings:

- [x] config validation via Zod (exported as a Node JS module for the unique reason of `entry.server.tsx` ..)
- [x] Awilix for dependency injection ( real advantage of it is not obvious in the first PR tbh)
- [x] better logger

This is a follow up of  https://github.com/allthingsweb-dev/allthingsweb/pull/3

Missive.js will arrive later, and then DIC will make more sense (or if we had tests)

### CODE REVIEW

File: website/.env.dist [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
-SENTRY_DSN=your_dsn
-SENTRY_AUTH_TOKEN=your_auth_token
-SENTRY_ORG=your_org
-SENTRY_PROJECT=your_project
-INNGEST_ENABLED=true
-INNGEST_SIGNING_KEY=your_signing_key
-INNGEST_EVENT_KEY=your_event_key
+
+# Optionals 
+# SESSION_SECRET=
```
Comment (andrelandgraf) [inline website/.env.dist:9]: I think it's required: `sessionSecret: z.string().min(1),`

File: website/app/entry.server.tsx [hunk 2]
```diff
@@ -5,15 +5,15 @@ import { createReadableStreamFromReadable } from '@remix-run/node';
 import { RemixServer } from '@remix-run/react';
 import { isbot } from 'isbot';
 import { renderToPipeableStream } from 'react-dom/server';
-import { env } from './modules/env.server';
+import config from '~/config.server';
```
Comment (andrelandgraf) [inline website/app/entry.server.tsx:8]: Nice, thanks!

File: website/app/modules/posthog/posthog.server.ts [hunk 3]
```diff
@@ -1,38 +1,38 @@
...
-/**
- * Define properties of custom analytics events here.
- * @example
- *  // First time registration for an event. Use event slug as distinct ID.
- * 'attendee registered': {
- *   attendee_id: string;
- *   event_name: string;
- *   event_id: string;
- *   type: 'website' | 'Luma';
- * };
```
Comment (andrelandgraf) [inline website/app/modules/posthog/posthog.server.ts:19]: RIP code comment 🗡️

**Review by andrelandgraf — APPROVED** [review]

File: website/app/container.server.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
+import { createContainer, InjectionMode, asFunction, asValue } from 'awilix';
```
Comment (andrelandgraf) [inline website/app/container.server.ts]: Could be `modules/dependencies-context` or `modules/dep-injection` or similar?

File: website/.env.dist (hunk 1 above)
Comment (Plopix) [reply website/.env.dist:9]: it is but it is set automatically if not provided.

`sessionSecret: process.env.SESSION_SECRET || 'local-session-secret-' + crypto.randomUUID(),`

File: website/app/container.server.ts (hunk 4 above)
Comment (Plopix) [reply website/app/container.server.ts]: moved it to `modules` ;-)

**Comment by Plopix** [comment]
I will let you merge it, make sure it's all working as I don't know all the website :) unless you did already then I will merge :)

**Pull request merged**